src/wp-includes/option.php

```diff
@@ -902,6 +911,25 @@ function update_option( $option, $value, $autoload = null ) {
 	return true;
 }
 
+/**
+ * Utility wrapper for update_option() that requires the old value to compare against
+ *
+ * @see update_option()
+ *
+ * @since 6.5.0
+ *
+ * @param string      $option    Name of the option to update. Expected to not be SQL-escaped.
+ * @param mixed       $value     Option value. Must be serializable if non-scalar. Expected to not be SQL-escaped.
+ * @param mixed       $old_value The option value retrieved with get_option() initially to compare against
+ *                               to avoid any intermittent changes to the option to update the option with an old value.
+ * @param string|bool $autoload  Optional. @see update_option(). Default null.
+ *
+ * @return bool True if the value was updated, false otherwise.
```

Comment:
```suggestion
 * @param string|bool $autoload  Optional. @see update_option(). Default null.
 * @return bool True if the value was updated, false otherwise.
```